doc/tutorials/objdetect/generalized_hough_ballard_guil/generalized_hough_ballard_guil.markdown

```diff
@@ -0,0 +1,177 @@
+Object detection with Generalized Ballard and Guil Hough Transform {#tutorial_generalized_hough_ballard_guil}
+==================================================================
+
+@tableofcontents
+
+@prev_tutorial{tutorial_traincascade}
+
+Goal
+----
+
+In this tutorial you will lern how to:
+
+- Use @ref cv::GeneralizedHoughBallard and @ref cv::GeneralizedHoughGuil to detect an object
+
+Example
+-------
+
+### What does this program do?
+
+1. Load the image and template
+
+![image](images/generalized_hough_mini_image.jpg)
+![template](images/generalized_hough_mini_template.jpg)
+
+2. Instantiate @ref cv::GeneralizedHoughBallard with the help of `createGeneralizedHoughBallard()`
+3. Instantiate @ref cv::GeneralizedHoughGuil with the help of `createGeneralizedHoughGuil()`
+4. Set the required parameters for both GeneralizedHough variants
+5. Detect and show found results
+
+Note:
+
+- Both variants can't be instantiated directly. Using the create methods is required.
+- Guil Hough is very slow. Calculating the results for the "mini" files used in this tutorial
+  takes only a few seconds. With image and template in a higher resolution, as shown below,
+  my notebook requires about 5 minutes to calculate a result.
+
+![image](images/generalized_hough_image.jpg)
+![template](images/generalized_hough_template.jpg)
+
+### Code
+
+The complete code for this tutorial is shown below.
+@include generalizedHoughTransform.cpp
+
+Explanation
+-----------
+
+### Load image, template and setup variables
+
+```c++
+//  load source images
+Mat image = imread("images/generalized_hough_mini_image.jpg");
```

Comment:
The  common OpenCV approach is not to hard-code examples in markdown, but include code snippets with doxygen notation. Code snippets in form of C++ files are check with compiler.
Multi-language example:
- tutorial https://raw.githubusercontent.com/opencv/opencv/4.x/doc/tutorials/core/basic_linear_transform/basic_linear_transform.markdown
- https://github.com/opencv/opencv/blob/4.x/samples/cpp/tutorial_code/ImgProc/BasicLinearTransforms.cpp

In your case you need only `@snippet` directive without language switch.